### DISCUSSION TITLE
[Kavita] Different display settings for cover art

### DISCUSSION THREAD

**Discussion by scare376** [Kareadita/Kavita#2629] [Ideas]
### Idea Description

From: takatak00

-- 11/15/2023 12:28 AM
By default the first volume cover art is always shown for your series in the library. After awhile I find my library looks the same and little boring. I know there is a feature to manually select a specific cover art which is great but a little time consuming when you have many series. There are 2 settings I would like to request.
1- Random cover art. Everytime you open Kavita, your library looks different!
2- Show cover art of the volume your currently reading.

Additional comments from:
   maknilsin: 
   Komga handles this really well. Especially for comics where it lets you configure the library to show the next non-read issue. Especially for On Deck and Continue Reading views it’s amazing.
  droglo:
  I would love to have an option to show the cover of the newest volume.
  takatak00:
  Show the cover art of the newest volume sounds great!

### Duration of Using Kavita

x

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by lyingloverr** [comment]
This is a nice feature that adds a lot of charm :D

**Reply by raub21** [reply]
Love to see this feature and if possible show the cover of the latest issue from the series! Thank you!

**Comment by MortenCB** [comment]
I would love to be able to show the image from latest issue.

**Comment by ghost** [comment]
This would be a nightmare for people using FMD2 to download manga as most of the Chapters just use a random page as the cover art, if this would to become reality, please make it an option that defaults to off

**Comment by majora2007** [comment]
How are you expecting this to work? On a series-by-series basis it can be set or at a library level? My understanding for this type of feature is for magazines and my understanding is magazines are likely to be in a separate series, so having at a library level makes the most sense.

**Reply by lyingloverr** [reply]
I think in a perfect world there is both options. Series to series basis and a library level. Me personally though if I had to pick one i would rather have it on a library level. I can only speak for myself though.

**Comment by raub21** [comment]
For my case, I mainly would like to implement it in magazines.

**Comment by sammy-ette** [comment]
I would enable this for my manga library. I would also suggest a 3rd option of displaying the cover for the **latest** volume (instead of currently reading volume). This way it can be easily noticed that a new volume for something got added?

**Comment by majora2007** [comment]
So revisiting this and here's my thoughts:
1. A library option that allows you to set the way the series cover functions:
  - Default: Selects from the first (or override)
  - Last: Last Volume (or Chapter) (note: You have vol 1-10, and chapter 23, the series cover will be chapter 23)
  - Latest: Order by date added then find Last from that (note: Chapters are added in bulk on a scan, so you don't know if chapter 2 or chapter 3 came first, hence why falling back to Latest)
  - Last Progress: Essentially what the current progress point is (note: this is expensive as if you read chapter by chapter you're re-generating covers in the background for each user reading. I'm not sure this is actually a good idea at runtime (but nightly can work)
2. If a series cover is locked (via override or just locked via some api) then no overriding will take place. 
3. This affects the series cover so you'll also see this on library detail page as well when browsing series. 

Let me know if this is a solution that meets expectations?

**Reply by raub21** [reply]
@majora2007. Yes please, I would like to be able to see the latest cover of the latest chapter or special added in the Home screen. Users can notice that there is a new issue of the chapter or special being added to the library!  Thank you very much!

**Reply by Gin-no-kami** [reply]
The last progress option is exactly what I am looking for. As someone who reads who volumes of manga/books, having the cover update on the "On Deck" row will make the system seem more dynamic. As opposed to current implementation, where you will always see the cover from the first volume forever.

**Reply by JJMcNugget** [reply]
The "Last Progress" option would be awesome. I want to see the current cover of the volume im reading. Right now it feels like no progress always seeing the same first volume.

**Comment by Vital-Fernandez** [comment]
I would like Kavita/Kavita+ to generate simple cover images for collections that belong to a series theme (now it is taking the cover from one book or the one you uploaded). For example in a science pdf library, it would be nice to have "notebook-like" cover image just with the name of the sub-collection/series. For example, algebra, statistics,mechanical enginieering... Or just a nice font with the name of the series and a different background color for each series.

**Comment by DMcWetty** [comment]
I'm not sure if this has been developed yet, but I'd like to have a per-library setting for the cover art of series to show the most recent in-progress or unread cover. So if I have a 3-book series, and I've read Book 1 and started Book 2, then Book 2's cover art would show on the series collection. I say per-library, because I likely wouldn't use this for my Manga collection. 

Ex: I just finished Ready Player One, but the "On Deck" book in the series (Ready Player Two) shows the cover art for Ready Player One. Hopefully that makes sense.

**Comment by maknilsin** [comment]
@majora2007 - I know there's been little chatter on this one in awhile, but curious if this is something that would get some consideration in the future? 

Also well aware you've got a lot on and are generally always planning new features so completely understand if this is not a priority.

**Reply by majora2007** [reply]
It looks like quite a few people have expressed interest in the feature. I'm still not 100% sure on the feature, hence why I haven't merged the open PR in yet. I'll give this another think, but it probably wont come with v0.9.1 release.

@Fesaa

**Reply by sk0mik** [reply]
+1 here too. I switched from Komga and would love to see these options implemented (at least something like "First unread else first" cover setting like Komga has) .

**Reply by timgonzales** [reply]
+1 from me as well. I came over from running a local/closed ecosystem with YACReaderLibrary and YACReader. The one feature I currently miss from that platform was its 'Continue Reading' section displaying the cover of whatever volume or issue you were reading on the home screen of its iOS/Android app. Would be perfect if that were an option for the On Deck section that could do the same, or maybe implementing the feature in a whole different section that I can add on the Kavita web app's home screen.

**Reply by DMcWetty** [reply]
Glad to hear you're giving this another look. I'm looking forward to v0.9.1 so I wouldn't want this feature request to delay that release. It can wait.

**Reply by AndyMatt** [reply]
I was just looking into something similar, I love the Idea of changing the cover art depending on the current read state

**Comment by romquenin** [comment]
Adding my vote for this feature. I have several magazine series in my library, and it would be really helpful if the series cover could automatically update to show the latest volume/issue rather than always defaulting to the first one. Right now, when a new issue is added, the homepage/library thumbnail still shows the very first issue, which makes it hard to tell at a glance that a new one has arrived.

A per-library setting (rather than per-series) would be especially useful for those of us with many magazine series, since updating each one manually isn't practical. The "LatestVolume" option mentioned earlier in this thread sounds like exactly the right approach. Thanks for considering this!
